### ISSUE TITLE
"publish" signaling path in trickle ice branch should be simplified

### ISSUE THREAD

**Issue by jnoring** [ging/licode#245]
The publish path for the trickle ice branch looks like:

1. Client sends publish message.
2. Server receives publish message, creates WebRtcConnection/DtlsTransport(s)/NiceConnection(s)
3. Server sends response of the client's ID
4. Client generates offer
5. Client sends offer
6. Server receives offer, starts generating candidates, generates answer
6.1. Server sends over any candidates, client caches them until it gets answer.
7. Servers returns Answer
8. Client calls setlocaldescription, setremotedescription
9. Client adds any remote candidates, sends any generated local candidates.
10. client/server forward on any candidates generated at this point

I believe it should be simplified to:

1. Client generates offer
2. Client sends publish message that includes offer, calls set local description
3. Server receives publish message, creates WebRtcConnection/DtlsTransport(s)/NiceConnection(s)
4. Server processes included offer, starts generating candidates, generates answer
4.1. Server sends over any candidates, client caches them until it gets answer
5. Server responds with answer
6. client calls setremotedescription
7. client sends any generated local candidates, adds any remote candidates its received
8. client/server forward on any candidates generated at this point

Basically, in the publish message should be an offer, and the response from that should return both answer (without candidates, of course) and stream ID

thoughts?

**Comment by IVBakker** [comment]
Not a bad idea but it would make a big change in the echange protocol already set. But not a bad idea

**Comment by IVBakker** [comment]
Not a bad idea. It would make change significantly the exchange protocol already set so it can be dangerous. Also there might - I said might - have some edge case that we don't see (so far I don't see any).

**Comment by jnoring** [comment]
Another simplification: rather than having both sides "trickle" candidates, it makes more sense to simply have licode return a full OFFER.  This is completely compliant with the trickle ICE spec, and candidate gathering server side is extremely fast.  That saves some additional overhead.

**Comment by francoisTemasys** [comment]
For the last point, I'm working on that on #255

**Comment by IVBakker** [comment]
For last point, see #255. Should be pushed tomorrow

**Comment by jnoring** [comment]
@IVBakker are you planning on implementing that as the spec dictates?  Because that'd also be really awesome.  In the SDP exchange there's supposed to be an ice-option field specifying if the client supports trickling candidates or not.

What would be great about this is (in theory) licode would support both the old signaling as well as trickle ICE.

**Comment by IVBakker** [comment]
@jnoring I just pushed that.
I haven't checked the option. I let the user set that on the webrtcconnection layer (in ErizoJS). It's when creating the object.
If the value is not defined, the MCU is NOT using iceTrickle for sending the offer and candidates.
